### PR TITLE
Taoshaf.rewrite so3 exp and log

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/so3.py [hunk 1]
```diff
@@ -84,10 +84,15 @@ def exp_map(tangent_vector: torch.Tensor) -> LieGroup:
         theta = torch.linalg.norm(tangent_vector, dim=1, keepdim=True).unsqueeze(1)
         theta2 = theta ** 2
         # Compute the approximations when theta ~ 0
-        sel = theta >= 0.005
-        a = torch.where(sel, theta.cos(), 8 / (4 + theta2) - 1)
-        b = torch.where(sel, theta.sin() / theta, 0.5 * a + 0.5)
-        c = torch.where(sel, (1 - a) / theta2, 0.5 * b)
+        small_theta = theta < 0.005
```
Comment (luisenp) [inline theseus/geometry/so3.py:87]: What's the motivation for this flip in the indexing? (asking only out of curiosity, this looks fine to me).

File: theseus/geometry/so3.py [hunk 2]
```diff
@@ -110,23 +115,26 @@ def _log_map_impl(self) -> torch.Tensor:
...
         # theta != pi
-        sel1 = 1 + cth > 1e-7
-        # Compute the approximation of theta / sin(theta) when theta ~ 0
-        scale1 = torch.where(
-            theta[sel1] >= 0.005, theta[sel1] / sth[sel1], 1 + sth[sel1] ** 2 / 6
+        not_near_pi = 1 + cth > 1e-7
+        # Compute the approximation of theta / sin(theta) when theta is near to 0
+        small_theta = theta[not_near_pi] < 5e-3
+        non_zero = torch.ones(1, dtype=self.dtype, device=self.device)
+        sth_nz = torch.where(small_theta, non_zero, sth[not_near_pi])
```
Comment (luisenp) [inline theseus/geometry/so3.py:122]: Maybe `small_theta_nz`?

**Review by luisenp — APPROVED** [review]
LGTM.

File: theseus/geometry/so3.py (hunk 1 above)
Comment (fantaosha) [reply theseus/geometry/so3.py:87]: Just for the consistency with SE2.

File: theseus/geometry/so3.py (hunk 2 above)
Comment (fantaosha) [reply theseus/geometry/so3.py:122]: `sth_nz` means `sine_non_zero`, not small `small_theta_non_zero`.

Comment (luisenp) [reply theseus/geometry/so3.py:122]: Ah, my bad, I just quickly read from `torch.where` and mixed up the args (although tbh `small_theta_nz` doesn't make much sense, I guess). 

Can you rename to something more explicit? It'll make it easier if someone less familiar with Lie Groups needs to read this later (maybe `sine_theta`, `cosine_theta`)?.

**Pull request merged**